### ISSUE TITLE
Task.of doesn't behave as expected

### ISSUE THREAD

**Issue by dgroch** [urbandrone/futils#1]
The API documentation gives this example

```
const {Task} = require('futils');

let one = Task.of((rej, res) => res(1));

one.run(
    (x) => console.error('Rejected: ' + x),
    (n) => console.log('Resolved: ' + n)
);

// logs "Resolved: 1"
```

When I run this example I get
```
Resolved: (rej, res) => res(1)
```

Seems my Task is not resolving a value but instead resolves a function with the same signature as the function Task is meant to accept

**Comment by urbandrone** [comment]
Sorry, my bad. Seems the example isn't correct and I have to fix it (alongside the explanation above which doesn't really explain much).

**tl;dr explanation**
Think of it like an alias for `Task.resolve` and use it the same way.

**Long explanation**
The `Task.of` method "lifts" a given value into a Task, so the example really should be:

```
const {Task} = require('futils');

let one = Task.of((rej, res) => res(1));

one.run(
    (x) => console.error('Rejected: ' + x),
    (n) => console.log('Resolve: ' + n()) // notice the invocation, because "n" is the function given to "of"
);
```

which could be replaced with this:

```
const {Task} = require('futils');

let one = Task.of(1); // without wrapper

one.run(
    (x) => console.error('Rejected: ' + x),
    (n) => console.log('Resolve: ' + n) // no wrapper, no invocation needed
);
```

Usually you will use `Task.of` if you already have a value but you want to place it into a Task for some reason. `of` allows you to do so without calling `new` together with the constructor version which would be a bit tedious. The example in the docs mixes these two things up.

Most of the time you want to use a Task to create a wrapper for asynchronous operations which indeed uses `new` and takes a function which accepts to other functions as arguments, for example:

```
const {Task} = require('futils');
const $ = require('jquery');

const getJSON = (url) => new Task((rej, res) => {
    $.getJSON(url).
     done((data, status, jqXhr) => res(data)). // resolve with data, ignore rest
     fail((jqXhr, status, err) => rej(err)); // reject the error
});

getJSON('url/to/config.json'). // get the config.json file
    map((json) => json.port).
    run(
        (err) => console.error(err.message), // if we have an error, we end up here
        (port) => console.log(port) // on success we get the port
    );
```

Now, what is `Task.of` good for? In the example you could wrap the URL to the `config.json` file into a Task and use it right away:

```
const {Task} = require('futils');
const $ = require('jquery');

const getJSON = (url) => new Task((rej, res) => {
    $.getJSON(url).
     done((data, status, jqXhr) => res(data)).
     fail((jqXhr, status, err) => rej(err));
});

const URL = Task.of('url/to/config.json');

URL.
    flatMap(getJSON).
    map((json) => json.port).
    run(
        (err) => console.error(err.message),
        (port) => console.log(port)
    );
```

Without using the `Task.of` method, you need to put the URL into a Task like this:

```
const {Task} = require('futils');
const $ = require('jquery');

const getJSON = (url) => ...;

const URL = new Task((rej, res) => res('url/to/config.json'));

URL.
    flatMap(getJSON).
    ...
```

If you want a more concrete example, think about user authentication. Maybe you could return a URL to get the next data from depending on the result of an authentication attempt. If the attempt is a success, a user gets his/her dashboard and if it fails he/she gets the login form again.

```
const {Task} = require('futils');
const {getXhr, postXhr} = require('./ajaxhelpers'); // assuming you already have these

const AuthUrl = 'url/to/authenticate';
const OkUrl = 'url/to/dashboard';
const FailUrl = 'url/to/login';

const validateAuthAttempt = ({id, isValid}) => isValid ? Task.of(`${OkUrl}/${id}`) : Task.of(FailUrl); 

postXhr(AuthUrl, {user: 'johndoe', password: 's0M3_p4$$word'}). // send credentials
    flatMap(validateAuthAttempt). // determine if credentials returned a valid id
    flatMap(getXhr). // get next data to display (dashboard or login form)
    run(
        (ajaxErr) => /* somehow handle errors which may occur in any of the requests */,
        (result) => /* somehow render the final result */
    );
```

I hope this clears things up.

**Comment by dgroch** [comment]
Certainly does - thank you for the explanation!

**Comment by urbandrone** [comment]
Thanks for spotting the issue!
